File: examples/python/parse_delivery_callback.py
from kalourmade_arkesel import parse_delivery_callback
from kalourmade_arkesel.exceptions import ValidationError

# Simulate an incoming GET callback for demonstration purposes.
query = {"sms_id": "abc-123", "status": "DELIVERED"}

result = parse_delivery_callback(query)
print(f"sms_id={result['sms_id']} status={result['status']}")

# Demonstrate the failure case: a malformed callback is rejected.
try:
    parse_delivery_callback({"sms_id": "abc-123"})
except ValidationError as e:
    print(f"Correctly rejected malformed callback: {e}")

File: examples/node/parseDeliveryCallback.ts
import { parseDeliveryCallback, ValidationError } from '@kalourmade/arkesel-sms';

// Simulate an incoming GET callback for demonstration purposes.
const query = { sms_id: 'abc-123', status: 'DELIVERED' };

const result = parseDeliveryCallback(query);
console.log(`sms_id=${result.smsId} status=${result.status}`);

// Demonstrate the failure case: a malformed callback is rejected.
try {
  parseDeliveryCallback({ sms_id: 'abc-123' });
} catch (e) {
  if (e instanceof ValidationError) {
    console.log(`Correctly rejected malformed callback: ${e.message}`);
  }
}
